pkg/loader/load.go

```diff
@@ -430,16 +430,19 @@ func (s *loaderImpl) execDMLs(dmls []*DML) error {
 
 // Run will quit when meet any error, or all the txn are drained
 func (s *loaderImpl) Run() error {
+	txnManager := newTxnManager(1024)
```

Comment:
1. We can't control the memory consumed by the buffers here even when the capacity is limited by the number of DMLs.
1. If we do want to limit memory consumption by the number of DMLs, we can do that by replacing `chan *Txn` with `chan X` where X only contains single DML or DDL, we don't need to implement something similar to a buffered channel.

Reply:
1. It's not easy to measure the memory consumed by a single DML. It seems that the main memory usage in one single DML is map `OldValues` and `Values`. So maybe we can count the sizeof these two maps?
2. If we use X contains only single DML or DDL, we also need to add `txn` to `batchManager` but not single DML or DDL. A successfully executed single DML doesn't mean the whole `txn` is successful.

Reply:
We can't avoid OOM by checking everywhere we might use memory (and making the program more complex), so the problem we need to solve is to lower the possibility, which I think can be solved by just setting smaller buffer sizes.

Reply:
The problem is that setting smaller buffer sizes will cause performance digression. If we just set buffer size of loader.input to 8 the Update Event wil drop from 7k+ to 3k+. 
![image](https://user-images.githubusercontent.com/21104942/65104032-a71d0f00-da02-11e9-82e8-5a44912eee6e.png)
If we change `default` in [loader](https://github.com/pingcap/tidb-binlog/blob/9f3c81683bb3428c4940611a6203288474d4aff0/pkg/loader/load.go#L456) to `case <-time.After(10 * time.Microsecond)` the situation will get better, but we still have performance digression. However, if the binlogs are very few, it takes more time for `load` to execute cached binlogs.
![image](https://user-images.githubusercontent.com/21104942/65422746-ec45a480-de39-11e9-8874-7d497d0c59fd.png)

Reply:
any better suggestions for @lichunzhu  ? @suzaku

Reply:
Could you please change `output:   make(chan MergeItem)` back to `output:   make(chan MergeItem, 10)` and try again?

Reply:
@suzaku 
It still has performance digression. It's the situation which we set `default` to `case <-time.After(10 * time.Microsecond)`.
![image](https://user-images.githubusercontent.com/21104942/65493165-e5c03700-dee4-11e9-96cf-f884059b0e76.png)